### ISSUE TITLE
[Predictions]: CallBack doesn't return in convert() when URLError occurs

### ISSUE THREAD

**Issue by ruiguoamz** [aws-amplify/amplify-ios#678]
**Describe the bug**
Using provisioned AWS Transcribe resource in an unsupported region and call `Amplify.Predictions.convert()`, CallBack which contains error message doesn't return.

**To Reproduce**
1. Create a Predictions Sample App
2. Follow https://docs.amplify.aws/lib/predictions/transcribe/q/platform/ios to provision `AWS Transcribe` resource in `eu-west-2` region
3. Call `Amplify.Predictions.convert()` to transcribe an audio file
4. The call back is not returning (neither success case nor failure case)

**Expected behavior**
Failure message should be returned

**Environment(please complete the following information):**
 - Amplify Framework Version: 1.0.4
 - Dependency Manager: Cocoapods
 - Swift Version : 5.0

**Device Information (please complete the following information):**
 - Device: Simulators
 - iOS Version: iOS

**Comment by ruiguoamz** [comment]
This might not be an issue anymore:
1. Amazon Transcribe is now available in four more regions according to this [link](https://aws.amazon.com/about-aws/whats-new/2019/01/amazon-transcribe-is-now-available-in-four-additional-regions/)
2. Backend resource fail to be provisioned in the rest of unsupported regions.

I tried to provision backend in unsupported region.

**eu-north-1(Stockholm)**
`amplify init` fails by showing up error message: 
```
CREATE_FAILED      predictionstranscription4509dade        AWS::CloudFormation::Stack Wed Nov 11 2020 22:46:47 GMT-0800 (Pacific Standard Time) Template error: Unable to get mapping for RegionMapping::eu-north-1::transcription


Resource is not in the state stackUpdateComplete
An error occurred during the push operation: Resource is not in the state stackUpdateComplete
```
**as-south-1(Cape Town), eu-south-1(Milan)**
`amplify push` fails by showing up error message like:
```
An error occurred when creating the CloudFormation stack
 Unable to get mapping for RegionMapping::ap-northeast-3::transcription
✖ Root stack creation failed
init failed
InvalidClientTokenId: The security token included in the request is invalid.
    at Request.extractError (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/protocol/query.js:50:29)
    at Request.callListeners (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'InvalidClientTokenId',
  time: 2020-11-12T16:52:33.854Z,
  requestId: '618c66b4-b147-4c36-9836-319869be963d',
  statusCode: 403,
  retryable: false,
  retryDelay: 49.850930757515414
}
```

**Comment by ruiguoamz** [comment]
On the time(07/29/2020) when I created this issue, provisioning AWS Transcribe backend resource in unsupported region was successful. But it fails now, for example in `eu-north-1(Stockholm)` region.

What I tried out is to provision Auth in `eu-north-1(Stockholm)` and put predictions configuration info in `amplifyconfiguration.json`, I can see `Connection 1: received failure notification`, `Connection 1: failed to connect 12:8, reason -1` and `Connection 1: encountered error(12:8)` printed out when calling `Amplify.Predictions.convert()`

I don't quite remember what message I saw four months ago. But I guess these messages might also showed up at that time.
